python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -215,6 +220,17 @@ export default function aiconfigReducer(
     case "ADD_PROMPT_AT_INDEX": {
       return reduceInsertPromptAtIndex(dirtyState, action.index, action.prompt);
     }
+    case "CLEAR_OUTPUTS": {
+      // This action happens before the server request is even sent; This is to maximize responsiveness of the client state.
+      // Once the server response comes in we can use the CONSOLIDATE_AICONFIG action to update the client with the response and that action would trigger a re-render.
+      return {
+        ...dirtyState,
+        prompts: dirtyState.prompts.map((prompt) => ({
+          ...prompt,
+          outputs: [],
+        })),
```

Comment:
Thinking on this a bit: should we do empty outputs array [] or just delete outputs entirely? If the former, just realized we should not do this for any prompt that doesn't have outputs defined because this will end up adding empty outputs array to the config for the prompt when it never had outputs key in the first place.

So maybe something like:
```
const prompts = state.prompts.map((prompt) => {
  if (prompt.outputs) {
    return {
       ...prompt,
        outputs: [] // or undefined
     }
  } else {
    return prompt;
  }
});

// If we want to remove instead of empty array, can do:
// for (const prompt in prompts) {
//   if (prompt.outputs) {
//    delete prompt.outputs;
//  }
// if we don't want to delete outputs here, just inline the prompts array below as before, but with the 
// map logic from above

return {
  ...dirtyState,
  prompts,
};
}
```

Reply:
I would say +1 to delete the outputs because as you pointed out, empty array is consistent with our design to have no outputs if outputs aren't defined

Reply:
left a comment on slack but to reiterate:

Makes sense to update here but the server side will still have prompt.outputs as [], even though client here might have it deleted.

For this change to be persisted, I see two options, which one should I do?

1. Update AIConfig sdk method delete_output to forcefully delete the attribute? Although I'm not sure if its expected to be [] anywhere right now.

2. Alternatively Delete the outputs field inside the server endpoint in api/clear_outputs and AIConfig sdk doesn't need to be touched (easy change, set field to None after calling delete_output)

Reply:
Should we remove them here on the client and just do the sdk change separately in a later PR? Then we won't need to come back here to change it later